app/uk/gov/hmrc/gform/core/parsers/ValueParser.scala

```diff
@@ -606,7 +610,7 @@ trait ValueParser extends RegexParsers with PackratParsers with BasicParsers {
       OptionDataValue.FormCtxBased(formCtx)
     } | alphabeticOnly ~ expr ^^ { case prefix ~ expr =>
       OptionDataValue.ExprBased(prefix, expr)
-    } | formatParserAlphabeticOnly ^^ (str => OptionDataValue.StringBased(str))
+    } | alphabeticWithSpacesAndCommas ^^ (str => OptionDataValue.StringBased(str))
```

Comment:
This change may cause forms to get broken. For example valid value like `op2` will be broken by this. Can we just not this change?

Reply:
I can update the regex to allow the conditions of the original word character matcher so it's like this:
`[a-zA-Z\s\d_,]`
Then commas and spaces will be allowed as well and rejected by the FormTemplateValidator. What do you think?

Reply:
From what I've looked at - If I don't change this regex, when the user adds non-expression values with spaces, the OptionData for the choice will be initialised as index based and will have no value property to check in FormTemplateValidator. So the user remains unaware of the issue. 
Do you have any thoughts on another way around this? As the current error (if you leave one value with a space by accident or otherwise) is the same as you mentioned:
`Choice component has some options with 'value' and some options without 'value'`

Reply:
Maybe throw exception like is done [here](https://github.com/hmrc/gform/blob/a4a45c0ba0df2a6337720ac908bf78c31311db13/app/uk/gov/hmrc/gform/sharedmodel/formtemplate/OptionDataValue.scala#L32)